wd_core.au3

```diff
@@ -1675,20 +1677,25 @@ EndFunc   ;==>__WD_CloseDriver
 ; #INTERNAL_USE_ONLY# ===========================================================================================================
 ; Name ..........: __WD_EscapeString
 ; Description ...: Escapes designated characters in string.
-; Syntax ........: __WD_EscapeString($sData)
-; Parameters ....: $sData - the string to be escaped
-; Return values..: Escaped string.
+; Syntax ........: __WD_EscapeString($sData[, $iOption = 0])
+; Parameters ....: $sData   - the string to be escaped
+;                  $iOption - [optional] Default is 0.
+; Return values .: Escaped string.
 ; Author ........: Danp2
 ; Modified ......:
-; Remarks .......:
+; Remarks .......: See $JSON_* constants in json.au3 for $iOption possibilities
 ; Related .......:
 ; Link ..........:
 ; Example .......: No
 ; ===============================================================================================================================
-Func __WD_EscapeString($sData)
-	Local $sRegEx = "([" & $_WD_ESCAPE_CHARS & "])"
-	Local $sEscaped = StringRegExpReplace($sData, $sRegEx, "\\$1")
-	Return SetError($_WD_ERROR_Success, 0, $sEscaped)
+Func __WD_EscapeString($sData, $iOption = 0)
+	If BitAND($iOption, $JSON_MLREFORMAT) Then
+		$sData = StringRegExpReplace($sData, '(?m)[\R\t]','') ; Strip tabs and CR/LFs
```

Comment:
`(?m)[\R\t]` is still wrong
try to use it with:
```js
return window.devicePixelRatio
```
in my testing I ended with `[\r\t]`

Reply:
I'm not sure what you mean by the above. Can you post a runnable snippet of code that demonstrates the issue?

FWIW, I was testing with this --

```autoit
#include "json.au3"

Local $IDX_Frame = 'test'
Local $s_CSS_Parser = '1 2 3 4'
Local $s_JavaScript = _
"	let style = document.createElement('style');									" & @CRLF & _
"	style.id = 'WEBDRIVER_CSSStyles-theme" & $IDX_Frame & "';							" & @CRLF & _
"	style.innerHTML = `" & $s_CSS_Parser & "`;										" & @CRLF & _
"	document.documentElement.insertBefore(style, document.body.nextSibling);		" & @CRLF & _
""
ConsoleWrite($s_JavaScript & @CRLF)
ConsoleWrite(__WD_EscapeString($s_JavaScript, $JSON_MLREFORMAT) & @CRLF)
```

Reply:
Sure.

Reply:
my testing code:
```autoit
#include "..\json.au3"
#include "..\wd_core.au3"

_Example()
Func _Example()
	Local $IDX_Frame = '1'
	Local $s_CSS_Parser = '* {background-color: white !important;}'
	Local $s_JavaScript = _
			"	let style = document.createElement('style');									" & @CRLF & _
			"	style.id = 'WEBDRIVER_CSSStyles-theme" & $IDX_Frame & "';							" & @CRLF & _
			"	style.innerHTML = `" & $s_CSS_Parser & "`;										" & @CRLF & _
			"	document.documentElement.insertBefore(style, document.body.nextSibling);		" & @CRLF & _
			""
	ConsoleWrite($s_JavaScript & @CRLF)
	ConsoleWrite('- TEST 1' & @CRLF)
	ConsoleWrite(__WD_EscapeString1($s_JavaScript, $JSON_MLREFORMAT) & @CRLF)
	ConsoleWrite('- TEST 2' & @CRLF)
	ConsoleWrite(__WD_EscapeString2($s_JavaScript, $JSON_MLREFORMAT) & @CRLF)

EndFunc   ;==>_Example

Func __WD_EscapeString1($sData, $iOption = 0)
	If BitAND($iOption, $JSON_MLREFORMAT) Then ; JavaScript linearization first
		$sData = StringRegExpReplace($sData, '(?m)[\R\t]', '')
	EndIf

	$sData = Json_StringEncode($sData, $iOption) ; JSON Escape Strings

	Return SetError($_WD_ERROR_Success, 0, $sData)
EndFunc   ;==>__WD_EscapeString

Func __WD_EscapeString2($sData, $iOption = 0)
	If BitAND($iOption, $JSON_MLREFORMAT) Then ; JavaScript linearization first
		$sData = StringRegExpReplace($sData, '[\r\t]', '')
	EndIf

	$sData = Json_StringEncode($sData, $iOption) ; JSON Escape Strings

	Return SetError($_WD_ERROR_Success, 0, $sData)
EndFunc   ;==>__WD_EscapeString
```

results:
```
	let style = document.createElement('style');									
	style.id = 'WEBDRIVER_CSSStyles-theme1';							
	style.innerHTML = `* {background-color: white !important;}`;										
	document.documentElement.insertBefore(style, document.body.nextSibling);		

- TEST 1
let style = document.createElement('style');\r\nstyle.id = '**WEBDIVE**_CSSStyles-theme1';\r\nstyle.innerHTML = `* {background-color: white !important;}`;\r\ndocument.documentElement.insertBefore(style, document.body.nextSibling);\r\n
- TEST 2
let style = document.createElement('style');\nstyle.id = '**WEBDRIVER**_CSSStyles-theme1';\nstyle.innerHTML = `* {background-color: white !important;}`;\ndocument.documentElement.insertBefore(style, document.body.nextSibling);\n

```

Reply:
so finally test it with:
```autoit
#include "..\json.au3"
#include "..\wd_core.au3"

_Example()
Func _Example()
	Local $IDX_Frame = '1'
	Local $s_CSS_Parser = '* {background-color: white !important;}'
	Local $s_JavaScript = _
			"	let style = document.createElement('style');									" & @CRLF & _
			"	style.id = 'WEBDRIVER_CSSStyles-theme" & $IDX_Frame & "';							" & @CRLF & _
			"	style.innerHTML = `" & $s_CSS_Parser & "`;										" & @CRLF & _
			"	document.documentElement.insertBefore(style, document.body.nextSibling);		" & @CRLF & _
			""
	ConsoleWrite($s_JavaScript & @CRLF)
	ConsoleWrite('- TEST 1' & @CRLF)
	ConsoleWrite(__WD_EscapeString1($s_JavaScript, $JSON_MLREFORMAT) & @CRLF)
	ConsoleWrite('- TEST 2' & @CRLF)
	ConsoleWrite(__WD_EscapeString2($s_JavaScript, $JSON_MLREFORMAT) & @CRLF)
	ConsoleWrite('- TEST 3' & @CRLF)
	ConsoleWrite(__WD_EscapeString3($s_JavaScript, $JSON_MLREFORMAT) & @CRLF)

EndFunc   ;==>_Example

Func __WD_EscapeString1($sData, $iOption = 0)
	If BitAND($iOption, $JSON_MLREFORMAT) Then ; JavaScript linearization first
		$sData = StringRegExpReplace($sData, '(?m)[\R\t]', '')
	EndIf

	$sData = Json_StringEncode($sData, $iOption) ; JSON Escape Strings

	Return SetError($_WD_ERROR_Success, 0, $sData)
EndFunc   ;==>__WD_EscapeString

Func __WD_EscapeString2($sData, $iOption = 0)
	If BitAND($iOption, $JSON_MLREFORMAT) Then ; JavaScript linearization first
		$sData = StringRegExpReplace($sData, '[\r\t]', '')
	EndIf

	$sData = Json_StringEncode($sData, $iOption) ; JSON Escape Strings

	Return SetError($_WD_ERROR_Success, 0, $sData)
EndFunc   ;==>__WD_EscapeString

Func __WD_EscapeString3($sData, $iOption = 0)
	If BitAND($iOption, $JSON_MLREFORMAT) Then ; JavaScript linearization first
		$sData = StringRegExpReplace($sData, '[\r\n\t]', '')
	EndIf

	$sData = Json_StringEncode($sData, $iOption) ; JSON Escape Strings

	Return SetError($_WD_ERROR_Success, 0, $sData)
EndFunc   ;==>__WD_EscapeString
```
my results:
```
--> SetHotKey () Restart failed. --> Press Ctrl+F11 to Stop.
	let style = document.createElement('style');									
	style.id = 'WEBDRIVER_CSSStyles-theme1';							
	style.innerHTML = `* {background-color: white !important;}`;										
	document.documentElement.insertBefore(style, document.body.nextSibling);		

- TEST 1
let style = document.createElement('style');\r\nstyle.id = 'WEBDIVE_CSSStyles-theme1';\r\nstyle.innerHTML = `* {background-color: white !important;}`;\r\ndocument.documentElement.insertBefore(style, document.body.nextSibling);\r\n
- TEST 2
let style = document.createElement('style');\nstyle.id = 'WEBDRIVER_CSSStyles-theme1';\nstyle.innerHTML = `* {background-color: white !important;}`;\ndocument.documentElement.insertBefore(style, document.body.nextSibling);\n
- TEST 3
let style = document.createElement('style');style.id = 'WEBDRIVER_CSSStyles-theme1';style.innerHTML = `* {background-color: white !important;}`;document.documentElement.insertBefore(style, document.body.nextSibling);
```

the third result is most accurate.

Reply:
I could have sworn that `(?m)[\R\t]` worked correctly when I tested it previously, but maybe I just overlooked the encoded @<!-->CRLF. \<shrug\> I think your `\r\n` is essentially the same as the `\v` I mentioned [here](https://github.com/Danp2/au3WebDriver/issues/465#issuecomment-1536456990).

Reply:
so it looks like this should be fine:
```autoit
Func __WD_EscapeString3($sData, $iOption = 0)
	If BitAND($iOption, $JSON_MLREFORMAT) Then ; JavaScript linearization first
		$sData = StringRegExpReplace($sData, '[\v\t]', '')
	EndIf

	$sData = Json_StringEncode($sData, $iOption) ; JSON Escape Strings

	Return SetError($_WD_ERROR_Success, 0, $sData)
EndFunc   ;==>__WD_EscapeString3
```